example.go

```diff
@@ -411,32 +408,52 @@ func getEnvVarOrExit(varName string) string {
 }
 
 // onErrorFail prints a failure message and exits the program if err is not nil.
-func onErrorFail(err error, message string) {
+func onErrorFail(err error, message string, a ...interface{}) {
 	if err != nil {
-		fmt.Printf("%s: %s\n", message, err)
+		fmt.Printf("%s: %s\n", fmt.Sprintf(message, a), err)
 		os.Exit(1)
 	}
 }
 
-func createClients(subscriptionID string, spToken *azure.ServicePrincipalToken) {
+func createClients(subscriptionID string, authorizer *autorest.BearerAuthorizer) {
+	sampleUA := fmt.Sprintf("Azure-Samples/virtual-machines-go-manage/%s", getCommit())
+
 	groupClient = resources.NewGroupsClient(subscriptionID)
-	groupClient.Authorizer = spToken
+	groupClient.Authorizer = authorizer
+	groupClient.Client.AddToUserAgent(sampleUA)
 
 	accountClient = storage.NewAccountsClient(subscriptionID)
-	accountClient.Authorizer = spToken
+	accountClient.Authorizer = authorizer
+	accountClient.Client.AddToUserAgent(sampleUA)
 
 	vNetClient = network.NewVirtualNetworksClient(subscriptionID)
-	vNetClient.Authorizer = spToken
+	vNetClient.Authorizer = authorizer
+	vNetClient.Client.AddToUserAgent(sampleUA)
 
 	subnetClient = network.NewSubnetsClient(subscriptionID)
-	subnetClient.Authorizer = spToken
+	subnetClient.Authorizer = authorizer
+	subnetClient.Client.AddToUserAgent(sampleUA)
 
 	addressClient = network.NewPublicIPAddressesClient(subscriptionID)
-	addressClient.Authorizer = spToken
+	addressClient.Authorizer = authorizer
+	addressClient.Client.AddToUserAgent(sampleUA)
 
 	interfacesClient = network.NewInterfacesClient(subscriptionID)
-	interfacesClient.Authorizer = spToken
+	interfacesClient.Authorizer = authorizer
+	interfacesClient.Client.AddToUserAgent(sampleUA)
 
 	vmClient = compute.NewVirtualMachinesClient(subscriptionID)
-	vmClient.Authorizer = spToken
+	vmClient.Authorizer = authorizer
+	vmClient.Client.AddToUserAgent(sampleUA)
+}
+
+func getCommit() string {
```

Comment:
I'm kinda thinking this should be moved to go-autorest so we can share it among the other samples.  You don't have to do it in this PR.